lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -575,6 +583,40 @@ def store_podspec(options = nil)
         end
       end
 
+      #--------------------------------------#
+
+      SCRIPT_PHASE_REQUIRED_KEYS = [:name, :script].freeze
+
+      SCRIPT_PHASE_OPTIONAL_KEYS = [:shell_path, :input_files, :output_files, :show_env_vars_in_log].freeze
+
+      ALL_SCRIPT_PHASE_KEYS = (SCRIPT_PHASE_REQUIRED_KEYS + SCRIPT_PHASE_OPTIONAL_KEYS).freeze
+
+      # Stores the script phase to add for this target definition.
+      #
+      # @param  [Hash] options
+      #         The options to use for this script phase. The required keys
+      #         are: `:name`, `:script`, while the optional keys are:
+      #         `:shell_path`, `:input_files`, `:output_files` and `:show_env_vars_in_log`.
+      #
+      # @return [void]
+      #
+      def store_script_phase(options)
+        option_keys = options.keys
+        unless option_keys.all? { |key| ALL_SCRIPT_PHASE_KEYS.include?(key) }
+          raise StandardError, 'Unrecognized options for the shell script ' \
```

Comment:
might be nice to say which options are unrecognized, and print out the known ones?